clang/test/Sema/matrix-type-operators.c

```diff
@@ -17,12 +17,12 @@ void add(sx10x10_t a, sx5x10_t b, sx10x5_t c) {
   // expected-error@-1 {{assigning to 'sx10x10_t' (aka 'float __attribute__((matrix_type(10, 10)))') from incompatible type 'sx5x10_t' (aka 'float __attribute__((matrix_type(5, 10)))')}}
 
   a = b + &c;
-  // expected-error@-1 {{invalid operands to binary expression ('sx5x10_t' (aka 'float __attribute__((matrix_type(5, 10)))') and 'sx10x5_t *' (aka 'float  __attribute__((matrix_type(10, 5)))*'))}}
-  // expected-error@-2 {{casting 'sx10x5_t *' (aka 'float  __attribute__((matrix_type(10, 5)))*') to incompatible type 'float'}}
+  // expected-error@-1 {{invalid operands to binary expression ('sx5x10_t' (aka 'float __attribute__((matrix_type(5, 10)))') and 'sx10x5_t *' (aka 'float * __attribute__((matrix_type(10, 5)))'))}}
```

Comment:
hi @fhahn!

Thanks for taking a look at this!

It was intentional, but this is the consequence of the change I commented on [here](https://github.com/llvm/llvm-project/pull/111415/files#r1790670980) hoping you'd weigh in on. I made the change this way in TypePrinter because it was more convenient to insert the HLSL printing style that way, but I can maintain the previous behavior. The reason I thought that perhaps the alternate position of `*` might be correct or preferable is because it is the way that vectors are printed with an admittedly contrived and double erroneous [example](https://godbolt.org/z/Wjzzx7sov).

At any rate, we can defer the discussion of whether the vector printing should conform with the matrix printing or vice versa for later. For now, I'll implement the printing so that it maintains previous behavior.

Reply:
Another oddity in how pointers to matrices were printed before was a double space between the type and the attribute. It's minor, but it suggested something was missing there.

Reply:
The new printing is illegal. You've changed it from a matrix of `float` to a matrix of `float*`. The type printer shouldn't be printing types in a way that changes their meaning.

Reply:
Right, then perhaps we should look into how vectors are printed.

Reply:
I created an issue for this: https://github.com/llvm/llvm-project/issues/114868 It's not really related to HLSL as we don't have pointers or references (yet). I have a tentative fix, but I'd like to see if anyone has thoughts on the further impact it will have on existing type printing.